src/app/product/product.component.html

```diff
@@ -0,0 +1,5 @@
+<div class="containers_row">
+  <mat-progress-spinner *ngIf="!product" mode="indeterminate"></mat-progress-spinner>
```

Comment:
is there need of mode ?

Reply:
There are tow types of mode determinate and indeterminate.
determinate works as per progress but indeterminate will work until timeout wouldn't happen.